src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -563,6 +612,12 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
                gradDensityTopOfEdge = (layerThicknessEdge(k - 1, iEdge)*gradDensityEdge(k - 1, iEdge) + &
                                        layerThicknessEdge(k, iEdge)*gradDensityEdge(k, iEdge))/(layerThicknessEdge(k - 1, iEdge) + &
                                                                                                 layerThicknessEdge(k, iEdge))
+               dzdxTopOfEdge = (layerThicknessEdge(k - 1, iEdge)*dzdxEdge(k - 1, iEdge) + &
+                                layerThicknessEdge(k, iEdge)*gradDensityEdge(k, iEdge))/(layerThicknessEdge(k - 1, iEdge) + &
```

Comment:
@vanroekel you are trying to simply interpolate `dzdxEdge` from middle to top of cell here, right? This `gradDensityEdge(k,iEdge)` should be `dzdxEdge(k,iEdge)`?